Aufgabe: Project (#279)
PM-Dungeon/PM-Lecture: markdown/assignments/pool_p/projekt/index.md

---
type: assignment
title: "Projekt"
author: "Andre Matutat (FH Bielefeld)"

> points: 20

Reviewer: Wolltest Du das nicht auf 15 Punkte runterdrehen? Zumindest hatte ich https://github.com/PM-Dungeon/PM-Lecture/pull/279#issue-1162757054 so verstanden?

Author: Ne wollte ich nicht. Ich wollte/habe die besteh Grenze von pool_p von 20 auf 15 verschoben.

Reviewer: ah, ok. habe ich wohl durcheinander gebracht.

> weight: 1

Reviewer: Das brauchen wir nicht, wenn es nur die eine Seite gibt. (Aber es schadet auch nicht.)

hidden: true
---

## Projekt

Während des Projektes sollen Sie sich ein größeres Feature für Ihren Dungeon überlegen, konzeptionieren und implementieren.

> Ein Projekt hat eine Laufzeit von 2 Wochen. In dieser Zeit bearbeiten Sie nur die Aufgaben aus dem pool_k und Ihr Projekt. Andere Pool-Aufgaben müssen Sie nicht parallel bearbeiten.

Reviewer: ```suggestion
Ein Projekt hat eine Laufzeit von **zwei** Wochen. In dieser Zeit bearbeiten Sie nur die Aufgaben aus dem `[Pool K]({{<ref "/assignments/pool_k" >}})`{=markdown} und Ihr Projekt. Andere Pool-Aufgaben müssen Sie nicht parallel bearbeiten.
```

> Ein Projekt hat eine Laufzeit von **zwei** Wochen. In dieser Zeit bearbeiten Sie nur die Aufgaben aus dem `[Pool K]({{<ref "/assignments/pool_k" >}})`{=markdown} und Ihr Projekt. Andere Pool-Aufgaben müssen Sie nicht parallel bearbeiten.

Reviewer: ```suggestion
Ein Projekt hat eine Laufzeit von **zwei** Wochen. In dieser Zeit bearbeiten Sie nur die Aufgaben aus dem `[Pool K]({{< ref "/assignments/pool_k" >}})`{=markdown} und Ihr Projekt. Andere Pool-Aufgaben müssen Sie nicht parallel bearbeiten.
```

> Ein Projekt hat eine Laufzeit von **zwei** Wochen. In dieser Zeit bearbeiten Sie nur die Aufgaben aus dem `[Pool K]({{< ref "/assignments/pool_k" >}})`{=markdown} und Ihr Projekt. Andere Pool-Aufgaben müssen Sie nicht parallel bearbeiten.

Reviewer: ```suggestion
Ein Projekt hat eine Laufzeit von **zwei** Wochen. In dieser Zeit bearbeiten Sie nur die Aufgaben aus dem `[Pool K]({{< ref "/assignments/pool_k" >}})`{=markdown} und Ihr Projekt. Andere Pool-Aufgaben können Sie nicht parallel bearbeiten.
```

Sie können insgesamt zwei Projekte durchführen, müssen aber mindestens eines durchführen.

> Um den pool_p erfolgreich zu absolvieren, müssen sie mindestens 15-Punkte holen.

Reviewer: ```suggestion
Um den `[Pool P]({{<ref "/assignments/pool_p" >}})`{=markdown} erfolgreich zu absolvieren, müssen sie hier **mindestens 15 Punkte** erreichen.
```

> Sie können die Projekte erst ab dem xx.xx (nach dem ersten E-Assesment) starten.

Reviewer: ```suggestion
Sie können die Projekte erst ab Woche 9 (d.h. ab KW22, also in der Woche nach dem ersten E-Assessment) starten.
```

> Um den `[Pool P]({{<ref "/assignments/pool_p" >}})`{=markdown} erfolgreich zu absolvieren, müssen sie hier **mindestens 15 Punkte** erreichen.

Reviewer: ```suggestion
Um den `[Pool P]({{< ref "/assignments/pool_p" >}})`{=markdown} erfolgreich zu absolvieren, müssen sie hier **mindestens 15 Punkte** erreichen.
```

> Sie müssen das Projekt vorher mit ??? absprechen. Bereiten Sie dafür ein kurzes Konzept vor und erklären Sie, was Sie machen möchten, wie Sie dabei vorgehen wollen und welche Probleme/Fragen es gibt. Melden Sie sich bitte frühzeitig.

Reviewer: - Was bedeutet "kurzes Konzept"? Wir sollten eine Seitenzahl angeben, beispielsweise 1..3 Seiten, Format PDF.
- Was bedeutet "melden Sie sich frühzeitig"? Wie früh ist früh genug? Bei wem sollen sie sich melden?

Reviewer: siehe https://github.com/PM-Dungeon/PM-Lecture/issues/89#issue-1138610219

Nachdem Ihre Projekt-Idee genehmigt wurde, beginnt die Bearbeitungszeit.

Nach der ersten Woche müssen Sie ein schriftliches Feinkonzept und einen ersten Grobentwurf der Implementierung vorstellen.

Nach der zweiten Woche stellen Sie das Ergebnis vor.

> Bei der Punktevergabe kommt es nicht nur darauf an, ob Sie Ihr Ziel erreicht haben, sondern auch darauf, ob Sie die Ihnen bekannten Maßnahmen zur Einhaltung der Softwarequalität genutzt haben (Testfälle, Logging, Code-Style). Auch Ihre schriftliche Ausarbeitung fließt in die Punktevergabe mit ein.

Reviewer: ```suggestion
Bei der Punktevergabe kommt es nicht nur darauf an, ob Sie Ihr Ziel erreicht haben, sondern auch darauf, ob Sie die Ihnen bekannten Maßnahmen zur Einhaltung der Softwarequalität genutzt haben (Testfälle, Logging, Code-Style). Auch die Qualität Ihrer schriftlichen Ausarbeitung fließt in die Punktevergabe mit ein.
```

> Bei der Punktevergabe kommt es nicht nur darauf an, ob Sie Ihr Ziel erreicht haben, sondern auch darauf, ob Sie die Ihnen bekannten Maßnahmen zur Einhaltung der Softwarequalität genutzt haben (Testfälle, Logging, Code-Style). Auch die Qualität Ihrer schriftlichen Ausarbeitung fließt in die Punktevergabe mit ein.

Reviewer: - zur "schriftlichen Ausarbeitung" fehlen noch die Kriterien: Was soll aufgeschrieben werden, in welchem Umfang ...
